src/main/java/com/epam/grid/engine/provider/job/slurm/SlurmJobProvider.java

```diff
@@ -222,24 +224,30 @@ private String[] makeSbatchCommand(final JobOptions options) {
         final Context context = new Context();
         context.setVariable(OPTIONS, options);
         context.setVariable(LOG_DIR, logDir);
-        context.setVariable(ARGUMENTS, CommandArgUtils.toEscapeQuotes(options.getArguments()));
         context.setVariable(ENV_VARIABLES, CommandArgUtils.envVariablesMapToString(options.getEnvVariables()));
+
+        if (options.isCanBeBinary()) {
+            context.setVariable(BINARY_COMMAND, CommandArgUtils.toEscapeQuotes(options.getCommand()));
+        } else {
+            context.setVariable(ARGUMENTS, CommandArgUtils.toEscapeQuotes(options.getArguments()));
+        }
         return commandCompiler.compileCommand(getProviderType(), SBATCH_COMMAND, context);
     }
 
     private void validateJobOptions(final JobOptions options) {
         if (!StringUtils.hasText(options.getCommand())) {
             throw new GridEngineException(HttpStatus.BAD_REQUEST, "Command should be specified!");
         }
-        if (options.getPriority() != null && options.getPriority() < 0) {
+        if (options.getPriority() != null && (options.getPriority() < 0 || options.getPriority() > MAX_SENT_PRIORITY)) {
             throw new GridEngineException(HttpStatus.BAD_REQUEST, "Priority should be between 0 and 4_294_967_294");
         }
+        if (options.isCanBeBinary() && !options.getArguments().isEmpty()) {
```

Comment:
as I understand it isn't possible now to run command like `echo Hello, World!`? since we don't allow ti specify `arguments` for binary command